test/perf_counters_gtest.cc

```diff
@@ -37,26 +39,25 @@ TEST(PerfCountersTest, NegativeTest) {
     return;
   }
   EXPECT_TRUE(PerfCounters::Initialize());
-  EXPECT_FALSE(PerfCounters::Create({}).IsValid());
-  EXPECT_FALSE(PerfCounters::Create({""}).IsValid());
-  EXPECT_FALSE(PerfCounters::Create({"not a counter name"}).IsValid());
+  EXPECT_TRUE(PerfCounters::Create({}).IsValid());
+  EXPECT_TRUE(PerfCounters::Create({""}).IsValid());
+  EXPECT_TRUE(PerfCounters::Create({"not a counter name"}).IsValid());
```

Comment:
why would this be a valid PerfCounters?

Reply:
In fact, one of the results of these changes is that we do not need that `IsValid()` method anymore. It was there to prevent access to a null pointer. With the new filtering logic that went in yesterday, we are just dropping the bad entries and the object is always in a good state afterwards. Semantically it looks wrong and I think we can drop that IsValid method altogether since the object now is always valid anyways.

Reply:
Maybe replace with `HasAllRequestedCounters` or something like that?

Reply:
Then we would have to keep a bool around in the object to reflect the fact that the object has now less counters than was requested. Apart from the tests, there is no real use for that from the code point of view. 

We could yes compare the new number of counters with the expected number of counters and check that some were indeed dropped?

Reply:
Ah, because you already warn the user if some counter names "didn't make it"? Then yes, for the tests only we can just do what you suggested.

Reply:
So in the current PR and in another one I placed just now to fix an edge case, it compares the number of current, accepted PMCs with the number of requested. As in 

```c++
  EXPECT_EQ(PerfCounters::Create({}).num_counters(), 0);
  EXPECT_EQ(PerfCounters::Create({""}).num_counters(), 0);
  EXPECT_EQ(PerfCounters::Create({"not a counter name"}).num_counters(), 0);
```